Code/JavaWrappers/csharp_wrapper/CMakeLists.txt

```diff
@@ -92,22 +98,22 @@ SWIG_LINK_LIBRARIES(RDKFuncs ${RDKit_Wrapper_Libs}
 INSTALL(TARGETS RDKFuncs
         DESTINATION ${CMAKE_CURRENT_SOURCE_DIR} )
 
-
-if(NOT MSVC)
-   # code adapted from the wrapper code for
-   # GDCM: http://gdcm.svn.sf.net/viewvc/gdcm/trunk/Wrapping/Java/CMakeLists.txt?view=markup
-   ADD_CUSTOM_COMMAND(
-     OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/RDKit2DotNet.dll
-
-     COMMAND ${CMAKE_COMMAND} -E make_directory swig_csharp
-
-     ## 1. run this custom command only after swig has been run.
-     COMMAND ${GMCS_EXE} -platform:${PLATFORM} -out:RDKit2DotNet.dll -t:library "swig_csharp/*.cs"
-       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-     DEPENDS "${swig_generated_file_fullname}"
-   )
-   ADD_CUSTOM_TARGET(RDKFuncsDLL ALL
-     DEPENDS RDKFuncs ${CMAKE_CURRENT_SOURCE_DIR}/RDKit2DotNet.dll
-     COMMENT "building mono dll"
-   )
-endif(NOT MSVC)
+ADD_CUSTOM_COMMAND(
```

Comment:
Question: It looks like the changes in this file make it so that the C# wrappers can only be built on windows (and possibly only with MSVC++). I guess it's unlikely that anyone is using the RDKit with mono, but is there a reason to not leave that possibility open? 

There's no need to get the new tests working there, but at least continuing to allow a functioning build seems not terrible.

Reply:
It will work on Linux and well as Windows as Dot Net Core is available on both platforms.  I've got the tests working on Ubuntu.  It should be possible to get it working on OSX also, but that will require updating the csproj files to copy the native C++ library.

I took out the mono build as that technology is now replaced by Dot Net Core, though I can easily put it in.

I have 2 concerns.  Firstly I used .net6 in the build files, but .net8 is the current version. I wonder if it might be better to update to .net8.

Secondly, you need to have dotnet 6 SDK and runtime installed.  A windows user developing for net framework may not have those tools or versions installed.  Perhaps, it would be best to have a RDK_CSHARP_TEST cmake definition to make this step optional.

Reply:
> 
> I have 2 concerns. Firstly I used .net6 in the build files, but .net8 is the current version. I wonder if it might be better to update to .net8.
> 
> Secondly, you need to have dotnet 6 SDK and runtime installed. A windows user developing for net framework may not have those tools or versions installed. Perhaps, it would be best to have a RDK_CSHARP_TEST cmake definition to make this step optional.

I will defer to you on this: whatever you think makes the most sense.